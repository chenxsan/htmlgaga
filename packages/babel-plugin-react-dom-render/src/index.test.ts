import { transformSync } from '@babel/core'
import plug from './index'
const opts = {
  plugins: [plug],
  configFile: false,
  compact: true,
}
test('should import createElement and ReactDOM', () => {
  const str = ''
  const result = transformSync(str, opts)
  if (!result) return
  expect(result.code).toContain(
    `import ReactDOM from"react-dom";import{createElement}from'react';`
  )
})
test('should throw with default class component', () => {
  const str = 'export default class App extends React.Component {}'
  expect(() => transformSync(str, opts)).toThrowError()
})
test('should hydrate', () => {
  const str = 'export default function App() {}'
  const result = transformSync(str, {
    ...opts,
    plugins: [
      [
        plug,
        {
          hydrate: true,
        },
      ],
    ],
  })
  if (!result) return
  expect(result.code).toContain('ReactDOM.hydrate')
})
test('should test existence of getStaticProps', () => {
  const str = `export default function App() {}`
  const result = transformSync(str, opts)
  if (!result) return
  expect(result.code).toContain('typeof getStaticProps')
})
