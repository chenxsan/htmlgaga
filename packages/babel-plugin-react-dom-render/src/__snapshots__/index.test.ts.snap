// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should import createElement and ReactDOM 1`] = `"import ReactDOM from\\"react-dom\\";import{createElement}from\\"react\\";const __frontmatter__=typeof frontmatter!=='undefined'?frontmatter:{};if(typeof getStaticProps!==\\"undefined\\"){(async function(){const data=await getStaticProps();const pageProps=Object.assign(__frontmatter__,data.props);if(false){import(\\"false\\").then(({default:VOID_JS_APP})=>{ReactDOM.render(createElement(VOID_JS_APP,{Component:a,pageProps:pageProps}),document.getElementById(\\"app\\"));});}else{ReactDOM.render(createElement(a,pageProps),document.getElementById(\\"app\\"));}})();}else{if(false){import(\\"false\\").then(({default:VOID_JS_APP})=>{ReactDOM.render(createElement(VOID_JS_APP,{Component:a,pageProps:__frontmatter__}),document.getElementById(\\"app\\"));});}else{ReactDOM.render(createElement(a,__frontmatter__),document.getElementById(\\"app\\"));}}"`;
